ga4gh/datamodel/__init__.py

```diff
@@ -187,3 +218,21 @@ def _scanDataFiles(self, dataDir, patterns):
             atexit.register(_cleanupHtslibsMess, indexDir)
         if numDataFiles == 0:
             raise exceptions.EmptyDirException(dataDir, patterns)
+
+    def getFileHandle(self, dataFile):
+        if dataFile in self.memo_table:
```

Comment:
Here, we're splitting the job of the cache between the cache class and the mixin class, with both the memoisation table and the cache being updated to implement the LRU cache. All the cache functionality should be encapsulated in the cache. The getFileHandle() is a good pattern here, and we should keep this API. However, this should be a direct call to the cache class, with no logic.

We need a singleton instance of PysamFileHandleCache, and we also need to be able to modify the `maxCacheSize` externally during configuration. So, I think the simplest option is to have a module level variable called ``fileHandleCache``, which is created at module load time. I.e, near the top of the module (just after the definition of PysamFileHandleCache I guess, we have
```python
# Global singleton to cache file handles.
fileHandleCache = PysamFileHandleCache()
```

Then, during `configure` we can call
```python
datamodel.fileHandleCache.setMaxCacheSize(app.config["CACHE_MAX_SIZE"])
```

This method here, should then just be 
```
return fileHandleCache.getFileHandle()
```

This encapsulate things nicely. What do we think? @dcolligan, how does this design seem to you?